### PR TITLE
[ISSUE #3579]make kafka exporter's producer's max size can be changed by config

### PR DESCRIPTION
**Pull request by tydhot**
The reason for this change comes from #3579

### CODE REVIEW

**Comment by tydhot** [comment]
cc: @open-telemetry/collector-approvers 
BTW, I think it would be a challenge to find out why big size msg can't be exported by kafka exporter. The sarama producer would not log any data about this reason.

**Comment by tydhot** [comment]
cc @bogdandrutu .
I think this pr need to be reviewed.  If msg bigger than 1m(default max size in sarama) get into the sarama producer in  kafka exporter , it will be rejected  and the msg will put it to queue again. If the count of such size msg is big, the kafka exporter's queue will be full of this.  And you have to modify the code to avoid this. So I think it is necessary to add this configuration。

File: exporter/kafkaexporter/config.go [hunk 1]
```diff
@@ -44,6 +44,9 @@ type Config struct {
 
 	// Authentication defines used authentication mechanism.
 	Authentication Authentication `mapstructure:"auth"`
+
+	// MessageMaxSize defines the max size of the message can be sent by sarama.
+	MessageMaxSize int `mapstructure:"message_max_size"`
```
Comment (bogdandrutu) [inline exporter/kafkaexporter/config.go:49]: What is the unit? Probably better to avoid "int" and use one fixed size like int32 or int64

Comment (tydhot) [reply exporter/kafkaexporter/config.go:49]: It is byte in sarama's config. Maybe int64 is better.

Comment (tydhot) [reply exporter/kafkaexporter/config.go]: @bogdandrutu I changed it to int64. But it seemed a little weird. Sarama's config is int so I need changed it to int again when I set it to sarama's config.

Comment (bogdandrutu) [reply exporter/kafkaexporter/config.go]: int is fine then :). Please comment at least about the size

Comment (tydhot) [reply exporter/kafkaexporter/config.go]: oh yes, thanks! Btw, I have changed the config `MessageMaxSize` to `MessageMaxSizeMb`. In our actual use scenarios, I find this more convenient.:)

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Merged the other PR

**Pull request closed without merging**